aten/src/ATen/native/TensorShape.cpp

```diff
@@ -121,6 +121,93 @@ Tensor repeat(const Tensor& self, IntList repeats) {
   return result;
 }
 
+static std::vector<int64_t> infer_size(IntList shape, int64_t numel) {
```

Comment:
What does this function do

Reply:
I think you're being pedantic, but it infers the size of the reshaped array. As the documentation for reshape says,

```
A single dimension may be -1, in which case it's inferred from the remaining
dimensions and the number of elements in :attr:`input`.
```